### ISSUE TITLE
Interpreting training results, YOLOv8 and Improved yolov8

### ISSUE THREAD

**Issue by zyx0320** [ultralytics/ultralytics#21334]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Andrew
Hi, I'm Andrew, I'm new to AI. I would like to ask for some guidance on my research for my master thesis with YOLOv8, I want to detect cracks in the road surface.
My dataset has 823 images.The ratio of training, testing, and validation is 1:1:1
I did custom training for yolov8, I used batch size 4, image size 640 and epoch 200 and trained in a Jupyter notebook.
In addition, the BiLevelRoutingAttention double-layer routing attention mechanism and CSPStage generalized feature pyramid network are introduced into yolov8, and the same way is used for training
I would like to know:
1.Why does my  results tables seem to fluctuate so much?
2.Does my confusion matrix and evaluation metrics look normal？
3.What to do to make the results of the training look consistent with the results of the scientific research？
In addition, I trained Yolov8 with and without loading the pre-trained model, and conducted 200 and 300 rounds of training on the improved Yolov8, and the following is the result graph of the training.
Thank you very much for your help.

<img width="1369" height="261" alt="Image" src="https://github.com/user-attachments/assets/49f5e7c5-030b-45c9-b4b9-9223a73b74f0" />

<img width="272" height="619" alt="Image" src="https://github.com/user-attachments/assets/28149392-dc57-4c25-b9ee-27677d0b0fec" />

<img width="256" height="746" alt="Image" src="https://github.com/user-attachments/assets/49c5136b-34ab-4a3f-a6b5-48c11f180d5b" />
yolov8 loads the pre-trained model with 200 epoch of training results

<img width="877" height="661" alt="Image" src="https://github.com/user-attachments/assets/e3303c3b-e22b-4f4e-bbad-b847d84be069" />

<img width="771" height="517" alt="Image" src="https://github.com/user-attachments/assets/d9c9d176-05c5-47b0-a9c7-eb2183838bce" />

<img width="642" height="637" alt="Image" src="https://github.com/user-attachments/assets/596d27e3-3425-4828-9790-0c877fa477ef" />

<img width="699" height="692" alt="Image" src="https://github.com/user-attachments/assets/6eadb729-a2a2-428e-b560-b55ed735c105" />

<img width="791" height="524" alt="Image" src="https://github.com/user-attachments/assets/1721ee7b-b013-4d16-8bb7-ee741d963829" />

<img width="779" height="521" alt="Image" src="https://github.com/user-attachments/assets/ce55c525-8349-4cb5-8038-638a41f57ac7" />

<img width="655" height="455" alt="Image" src="https://github.com/user-attachments/assets/671ff2b8-c512-479a-a4c1-8adc26f26e06" />

<img width="621" height="310" alt="Image" src="https://github.com/user-attachments/assets/b9c44f42-5c2a-48c2-a294-2c9f844fe5bb" />

yolov8  does not loads the pre-trained model with 200 epoch of training results

<img width="567" height="376" alt="Image" src="https://github.com/user-attachments/assets/c2defffa-c2d7-4fa4-a674-5a63c95394f6" />

<img width="776" height="522" alt="Image" src="https://github.com/user-attachments/assets/aebcb02e-0623-4a3b-8d1b-215513ee4042" />

<img width="669" height="457" alt="Image" src="https://github.com/user-attachments/assets/5c1c0ff8-69e4-45e9-87de-38b13d403be3" />

<img width="839" height="425" alt="Image" src="https://github.com/user-attachments/assets/e8c2b29e-2ec1-42d2-bec7-ecae14bbfd9b" />

Improved Yolov8 loading pre-trained model 200epoch training 

<img width="721" height="359" alt="Image" src="https://github.com/user-attachments/assets/1280ad76-86d4-414d-bb8d-b7ae9ad9be14" />
Improved Yolov8 loading pre-trained model 300epoch training 
<img width="850" height="424" alt="Image" src="https://github.com/user-attachments/assets/2b7f1908-fe75-4634-8fbc-2a462e2b83b7" />

Improved Yolov8 training without loading pre-trained model 200epoch 

<img width="734" height="364" alt="Image" src="https://github.com/user-attachments/assets/f322f5f3-ebab-46ce-acd4-dd1d43938512" />

**Comment by UltralyticsAssistant** [comment]
👋 Hello @zyx0320, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

---

This is an automated response to help guide you to relevant resources and best practices. An Ultralytics engineer will review your post and assist you soon! 🛠️

**Comment by glenn-jocher** [comment]
Hi Andrew! Your training fluctuations are typical for small datasets (823 images) with batch size 4 - this creates noisy gradients and unstable training. For research consistency, I'd recommend: 1) Run multiple training sessions with different random seeds and report mean/std results, 2) Use larger batch sizes if GPU memory allows (8-16+), 3) Implement early stopping and learning rate scheduling, 4) Consider data augmentation to effectively increase dataset size. Your metrics look reasonable but the high variance suggests overfitting - try reducing model complexity or increasing regularization. For scientific rigor, focus on statistical significance across multiple runs rather than single best results. Check our [Model Training Tips guide](https://docs.ultralytics.com/guides/model-training-tips/) for optimization strategies.

**Comment by zyx0320** [comment]
> 嗨，安德鲁！对于批量大小为 4 的小型数据集（823 张图像），您的训练波动是典型的 - 这会产生嘈杂的梯度和不稳定的训练。为了研究的一致性，我建议： 1） 使用不同的随机种子运行多个训练会话并报告平均值/标准结果， 2） 如果 GPU 内存允许 （8-16+），则使用更大的批量， 3） 实施早期停止和学习率调度， 4） 考虑数据增强以有效增加数据集大小。您的指标看起来合理，但高方差表明过度拟合 - 尝试降低模型复杂性或增加正则化。为了科学的严谨性，请关注多次运行的统计显著性，而不是单个最佳结果。查看我们的[模型训练技巧指南](https://docs.ultralytics.com/guides/model-training-tips/)，了解优化策略。

Thank you for your reply, I will adjust according to your suggestion, and will communicate with you in time if there is a new follow-up

**Comment by glenn-jocher** [comment]
You're welcome Andrew! Best of luck with your master's thesis research on crack detection. Feel free to open new issues or join our discussions if you need further assistance as you implement these improvements.

**Comment by zyx0320** [comment]
> You're welcome Andrew! Best of luck with your master's thesis research on crack detection. Feel free to open new issues or join our discussions if you need further assistance as you implement these improvements.

I updated the dataset to 2515 images based on your suggestion, and increased the training rounds to 300, with a batch size of 8. Now the mAP50 and mAP50-95 images look normal, but the precision and recall images still do not look very good. I would like to ask what could be the reason for this? Is there any solution? Thank you for your answer. 
<img width="1119" height="555" alt="Image" src="https://github.com/user-attachments/assets/bd54d9f4-31f8-4d26-9d1c-695a6b5a3c89" />

**Comment by glenn-jocher** [comment]
Great progress Andrew! The improved mAP curves show your larger dataset is helping. Noisy precision/recall curves while mAP stabilizes often indicates confidence threshold sensitivity - your model finds cracks but with varying certainty. Try implementing learning rate scheduling with the `lrf` parameter (e.g., `lrf=0.01`) to reduce oscillations, and consider early stopping with `patience=50` to prevent overfitting. For crack detection specifically, class imbalance is common - you might benefit from focal loss or weighted sampling if available in your custom implementation.

**Comment by zyx0320** [comment]
Hello, thanks for your answer, I have a new problem, when I use the yolov8 raw yaml file to train the dataset, the result is as shown in the image below, what is the reason for this situation? My batch size,epoch, learning rate, etc. are the same as above, optimizer=SGD

<img width="692" height="340" alt="Image" src="https://github.com/user-attachments/assets/89d161c4-2d99-4cec-b3a4-332809d6928d" />

**Comment by glenn-jocher** [comment]
Looking at your training curves, this appears to be a configuration or data loading issue when switching to the raw YAML setup. Common causes include: incorrect dataset paths in the YAML file, different data preprocessing settings, or mismatched class configurations. Double-check your YAML file has the correct `path`, `train`, `val`, and `names` entries pointing to your updated 2515-image dataset. Also verify the `nc` (number of classes) matches your dataset exactly. Try printing your first few batches to ensure data is loading correctly - inconsistent results often stem from data path misconfigurations rather than training parameters.
